src/core/namespace.c

```diff
@@ -150,6 +154,14 @@ static const BindMount protect_system_strict_table[] = {
         { "/root",               READWRITE,    true  },      /* ProtectHome= */
 };
 
+/* If RootDirectory= is used, let's mount /sys and /proc into the it, but only as a fallback if the user hasn't mounted
+ * something there already. These mounts are hence overriden by any other explicitly configured mounts. */
+static const BindMount root_directory_table[] = {
+        { "/proc",               PROCFS,       false },
+        { "/dev",                BIND_DEV,     false },
+        { "/sys",                SYSFS,        false },
```

Comment:
Well, my assumption was that an environment without them is hardly useful: a massive chunk of the Linux API is below these directories. Now, I am fully away that this might clash with whatever people might already set up there, that's why the code is actually written so that it detects whether there's already a mount point in place and then skips our setup. And that should provide people with all the flexibility they need. all they have to do is mark the dirs as bind mounts (for example by bind mounting them on themselves). It's not a nice way out, but given that this is I am sure is a nichey corner case I thin it's OK.

(That said, maybe I am wrong and this will break for some people. If so, we can certainly revisit this, and maybe find a different scheme here too. The thing is simply though that I think to make RootDirectory= useful we really should mount these dirs automatically, since everything is just a nightmare to use for almost everybody. Hope that makes sense.